Resetpassword

noodlemoodle opened the pull request:
update resetpassword to reseetpasswordbytoken and bycode

noodlemoodle committed:
update resetpassword models

noodlemoodle committed:
update emailverificationcode

noodlemoodle committed:
update emailverificationservice

noodlemoodle committed:
add resetpasswordservice

noodlemoodle committed:
update resetpassword model

noodlemoodle committed:
add showsubmitaction

noodlemoodle committed:
add emaildisabled boolean

noodlemoodle committed:
Merge branch 'development' into resetpassword

jlhughes commented on src/foam/nanos/auth/resetPassword/ServerResetPasswordService.js:
```diff
@@ -0,0 +1,44 @@
+          x = x.put("subject", subject);
+  
+          DAO userDAO = (DAO) x.get("localUserDAO");
+          User user = (User) userDAO.find(EQ(foam.nanos.auth.User.EMAIL, newPasswordObj.getEmail()));
```
How does this system work for non-unique emails?

lchanmann committed:
Prevent resetting the original context

this happens for context that is a proxy of OrX, when calling put the OrX was replaced by a new OrX and the proxyX carries on with the new OrX

lchanmann committed:
initial MutableX

lchanmann committed:
Revert "Prevent resetting the original context"

This reverts commit 5cdcf7e3993dffa3b72cfc4a0f036192a37818a9.

lchanmann committed:
Fix tests

thepinkcacao committed:
Migrate test wizard changes to jrl

lchanmann committed:
Fix null effectiveUser throwing "Unknown User" exception

stacktrace:
2022-10-07T14:08:15.444-0400,qtp699221219-187,ERROR,UserCapabilityUpdateInterceptDAO,java.lang.RuntimeException: Unknown user
	at foam.util.Auth.sudo(Auth.java:60)
	at foam.nanos.crunch.ServerCrunchService.updateJunctionFor(ServerCrunchService.java:462)
	at foam.nanos.crunch.UserCapabilityUpdateInterceptDAO.put_(UserCapabilityUpdateInterceptDAO.java:111)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.approval.ApprovableAwareDAO.put_(ApprovableAwareDAO.java:889)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.ruler.RulerDAO.put_(RulerDAO.java:2051)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.CreatedAwareDAO.put_(CreatedAwareDAO.java:44)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.CreatedByAwareDAO.put_(CreatedByAwareDAO.java:46)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.LastModifiedAwareDAO.put_(LastModifiedAwareDAO.java:43)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.LastModifiedByAwareDAO.put_(LastModifiedByAwareDAO.java:48)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.AuthorizationDAO.put_(AuthorizationDAO.java:283)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.nanos.auth.PermissionedPropertyDAO.put_(PermissionedPropertyDAO.java:121)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.dao.ProxyDAO.put_(ProxyDAO.java:466)
	at foam.dao.AbstractDAO.put(AbstractDAO.java:674)
	at foam.nanos.dig.drivers.DigFormatDriver.daoPut(DigFormatDriver.java:704)
	at foam.nanos.dig.drivers.DigFormatDriver.put(DigFormatDriver.java:530)
	at foam.nanos.dig.DigWebAgent.execute(DigWebAgent.java:59)
	at foam.nanos.http.HttpParametersWebAgent.execute(HttpParametersWebAgent.java:234)
	at foam.nanos.pm.PMWebAgent.execute(PMWebAgent.java:34)
	at foam.nanos.http.ProxyWebAgent.execute(ProxyWebAgent.java:193)
	at foam.nanos.http.AuthWebAgent.execute(AuthWebAgent.java:89)
	at foam.nanos.http.NanoRouter.service(NanoRouter.java:127)

kgrgreer committed:
Remove resize listener onDetach in EditColumnsView.

sarthak-marwaha committed:
detach open listeners

AlexeyGreer committed:
Added new tab style and changed some colors

Commit:
Auto-next if first screen is invisible

jlhughes committed:
SMTPEmailService - ensure cron processes in created order

kgrgreer committed:
Make CommonColumnHandler a Singleton.

kgrgreer committed:
Formatting.

kgrgreer committed:
Cleanup.

kgrgreer committed:
Formatting.

AlexeyGreer committed:
Adressed Comments

kgrgreer committed:
Fix synchronization issue in MDAO addIndex.

jlhughes committed:
add missing , in requires block

kgrgreer committed:
Add x and MLang as global variables to make console scripting easier.

kgrgreer committed:
Fix row limit bug caused by QueryCachingDAO populating the wrong region of cache.

sarthak-marwaha committed:
Remove dollar sign

noodlemoodle committed:
rename model -> data, set factory on modelCls_ to return based on mode

thepinkcacao committed:
Bank account capability not granted bugfix on cards onboarding

kgrgreer committed:
Add an example of using tableColumns.

noodlemoodle committed:
fix cardselectionview when !choice[0] (#2116)

noodlemoodle committed:
remove redundant requires

noodlemoodle committed:
undo commentedout validate

noodlemoodle committed:
add alternateflowwao

noodlemoodle committed:
wrap alternateflowwizardlet wao with alternateeflowwao

noodlemoodle committed:
set isinaltflow

noodlemoodle committed:
update

sarthak-marwaha committed:
re populate filters when only one filter is set (#2120)

jlhughes committed:
Health - add memory used and use memory total rather than memory max (#2121)

jlhughes committed:
health xml - output default values (#2122)

* XML - add setter for  outputting default values and enable in HealthWebAgent to report memory used

* setter - follow existing builder pattern

* correct syntax error

kgrgreer committed:
Formatting.

jlhughes committed:
EmailService - add rateLimit (#2125)

sarthak-marwaha committed:
Status Page border (#2123)

* use css tokens in loading spinner

* Add StatusPageBorder

kgrgreer committed:
Remove ConstantSlotMultitonRefinement.

kgrgreer committed:
Cleanup.

jlhughes committed:
Medusa compaction (#2059)

Medusa Compaction, Performance, Replay enhancements.

jlhughes committed:
ReplayingInfo.count - refactor conflicting implementations (#2127)

* ReplayingInfo.count - refactor coinflicting implementations

* ReplayNode - remove commented code

kgrgreer committed:
Catch expected exceptions to avoid unnecessary warnings.

lchanmann committed:
Add rule is enabled check (#2128)

jlhughes committed:
Alarm - add reason to output.  Also conditionally output 'cleared' in emailTemplate to avoid warning that variable not provided (#2130)

smirnovaka committed:
Add months day mlang (#2126)

* add mlang for month and day

* cleanup

* add docs to month and day mlangs

kgrgreer committed:
Formatting.

jlhughes committed:
Medusa Performance tweaks (#2131)

* socket - refactor use of SessionServerBox from 'new' on each message to ThreadLocal reuse - afforded a 15% throughput increase

* PingCmd - refactor from constant to model which can capture echo time

* pm/om - adjust pm and om generation to better understand hot spots

* test for zero time and avoid divide by zero

* ClusterConfig - remove replayingInfo property, setup, and maintenance - another 5% increase

jlhughes committed:
cmd_ - comment out - breaking materialized views
